site_scons/site_tools/go_builder.py

```diff
@@ -3,20 +3,32 @@
 import subprocess  # nosec B404
 import os
 import re
+import json
 
 from SCons.Script import Configure, GetOption, Scanner, Glob, Exit, File
 
 GO_COMPILER = 'go'
 MIN_GO_VERSION = '1.16.0'
 include_re = re.compile(r'\#include [<"](\S+[>"])', re.M)
-GOINC_PREFIX = '"github.com/daos-stack/daos/src/'
 
 
-def _scan_go_file(node, _env, _path):
+def _scan_go_file(node, env, _path):
     """Scanner for go code"""
-    contents = node.get_contents()
+
     src_dir = os.path.dirname(str(node))
     includes = []
+    rc = subprocess.run([env.d_go_bin, 'list', '--json', os.path.basename(str(node))],
+                        cwd=src_dir, capture_output=True, check=True)
```

Comment:
Have you timed a build with this and without? Unless I'm misunderstanding, this will call `go list --json` for every single .go file in our tree, right? Seems like a lot of overhead.

Reply:
that was a concern.  It does however only call it for files that have changed so even if the first build is a little slower subsequent ones will see little difference.

Reply:
Ah, it doesn't work in CI as this is causing it to download some dependencies.  Needs more thinking about.

Reply:
Ok, I got this working.  Looking at the Jenkins logs there doesn't seem to be a measurable difference.